### PR TITLE
scantailor-universal: init at 0.2.14

### PR DESCRIPTION
**Pull request by unclamped**
## Description of changes
Adds [scantailor-universal](https://github.com/trufanov-nok/scantailor-universal), a fork based on Enhanced+Featured+Master versions of ScanTailor to the pkgs

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/graphics/scantailor/universal.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      sha256 = "sha256-n8NbokK+U0FAuYXtjRJcxlI1XAmI4hk5zV3sF86hB/s=";
+  };
+
+  buildInputs = [ qtbase zlib libjpeg_turbo libpng libtiff boost libcanberra openjpeg ];
+  nativeBuildInputs = [ cmake wrapQtAppsHook qttools ];
+
+  meta = with lib; {
+      description = "Interactive post-processing tool for scanned pages";
+      homepage = "https://github.com/trufanov-nok/scantailor";
+      license = licenses.gpl3;
```
Comment (azahi) [inline pkgs/applications/graphics/scantailor/universal.nix:35]: https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206

File: pkgs/applications/graphics/scantailor/universal.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+stdenv.mkDerivation rec {
+  pname = "scantailor-universal";
+  version = "0.2.14";
+
+  src = fetchFromGitHub {
+      owner = "trufanov-nok";
+      repo = pname;
+      rev = version;
+      fetchSubmodules = true;
+      sha256 = "sha256-n8NbokK+U0FAuYXtjRJcxlI1XAmI4hk5zV3sF86hB/s=";
```
Comment (azahi) [inline pkgs/applications/graphics/scantailor/universal.nix:26]:
```suggestion
      hash = "sha256-n8NbokK+U0FAuYXtjRJcxlI1XAmI4hk5zV3sF86hB/s=";
```

**Review by azahi — CHANGES_REQUESTED** [review]
Welcome to Nixpkgs. We use 2 spaces for indentation, so please fix that. Please also format the module inputs section like it's done in other packages, i.e. `, mkDerivation` (comma comes first). If you are looking for a formatting tool, you can use [nixpkgs-fmt](https://github.com/nix-community/nixpkgs-fmt).

**Comment by unclamped** [comment]
> We use 2 spaces for indentation, so please fix that.
Dangit! I was 100% sure I had fixed that. My bad.

As for the rest of the changes, do I fix each thing in its own commit or do I just end up squashing them all into the original commit? Also, do I mark you as a co-author of the changes or not?

**Comment by azahi** [comment]
> As for the rest of the changes, do I fix each thing in its own commit or do I just end up squashing them all into the original commit? 

Yes, it's the preferred way, unless you need to split up a big PR.

> Also, do I mark you as a co-author of the changes or not?

No, that's not necessary.

File: pkgs/applications/graphics/scantailor/universal.nix
Comment (azahi) [inline pkgs/applications/graphics/scantailor/universal.nix:1]: This file is still incorrectly formatted.

File: pkgs/applications/graphics/scantailor/universal.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+, qtbase
+, qttools
+, wrapQtAppsHook
+, zlib
+, openjpeg
+, libjpeg_turbo
+, libpng
+, libtiff
+, boost
+, libcanberra }:
```
Comment (azahi) [inline pkgs/applications/graphics/scantailor/universal.nix:15]:
```suggestion
, libcanberra 
}:
```

**Review by azahi — CHANGES_REQUESTED** [review]

File: pkgs/applications/graphics/scantailor/universal.nix (hunk 3 above)
Comment (unclamped) [reply pkgs/applications/graphics/scantailor/universal.nix:15]: :smiling_face_with_tear:

**Comment by unclamped** [comment]
alright I actually used the formatting tool this time, thought I wouldn't need it because I thought that was the only change left to do, oops

File: pkgs/applications/graphics/scantailor/universal.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+  buildInputs = [ qtbase zlib libjpeg_turbo libpng libtiff boost libcanberra openjpeg ];
+  nativeBuildInputs = [ cmake wrapQtAppsHook qttools ];
+
+  meta = with lib; {
+    description = "Interactive post-processing tool for scanned pages";
+    homepage = "https://github.com/trufanov-nok/scantailor";
+    license = licenses.gpl3Plus;
+    maintainers = with maintainers; [ unclamped ];
+    platforms = platforms.unix;
```
Comment (azahi) [inline pkgs/applications/graphics/scantailor/universal.nix:38]: Almost there. Only one nitpick left.

```suggestion
    platforms = platforms.unix;
    mainProgram = "scantailor-universal-cli";
```

**Review by azahi — CHANGES_REQUESTED** [review]

**Comment by unclamped** [comment]
image
should be all good

**Comment by azahi** [comment]
Result of `nixpkgs-review pr 260983` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>scantailor-universal</li>
  </ul>
</details>

**Review by azahi — APPROVED** [review]

**Pull request merged**